src/mixer/playerinfo.cpp

```diff
@@ -151,16 +161,21 @@ void PlayerInfo::updateCurrentPlayingDeck() {
             maxVolume = dvol;
         }
     }
-    if (maxDeck != m_currentlyPlayingDeck) {
-        m_currentlyPlayingDeck = maxDeck;
-        locker.unlock();
+    locker.unlock();
+
+    int oldDeck = m_currentlyPlayingDeck.fetchAndStoreRelease(maxDeck);
+    if (maxDeck != oldDeck) {
         emit currentPlayingDeckChanged(maxDeck);
-        emit currentPlayingTrackChanged(getCurrentPlayingTrack());
+        TrackPointer track;
+        if (maxDeck >= 0) {
+            track = getTrackInfo(PlayerManager::groupForDeck(maxDeck));
+        }
+        emit currentPlayingTrackChanged(track);
     }
 }
 
 int PlayerInfo::getCurrentPlayingDeck() {
-    QMutexLocker locker(&m_mutex);
+    updateCurrentPlayingDeck();
```

Comment:
Hiding this heavy-weight mutable operation inside a"getter" function that should actually be idempotent is a code smell. Do we have other options?

Reply:
This change was required to allow fast play/pause toggle via mpris. The 2 sec timer based update is too long. 
Would it be OK to rename the function to calcCurrentPlayingDeck()?

Reply:
updateCurrentPlayingDeck() could also return the deck number/index. The getter should be read-only.

Reply:
I have reverted the this.